lib/traject/macros/bibtex.rb

```diff
@@ -9,28 +11,77 @@ module BibTeX
+        accumulator << record.key.to_s
+      end
+    end
+
+    # raw serialization of BibTeX::Entry
+    def extract_bibtex_raw
+      lambda do |record, accumulator, _context|
+        accumulator << record.to_s
+      end
+    end
+
+    def extract_bibtex_publication
+      lambda do |record, accumulator, _context|
+        %i(journal publisher).each do |key|
+          accumulator << record.send(key).to_s.presence if record.respond_to?(key)
+        end
+      end
+    end
+
+    # formatted BibTeX::Entry in Chicago style as HTML
+    def extract_bibtex_formatted_bibliography
+      lambda do |record, accumulator, _context|
+        html = Bibliography.new(record.to_s).to_html
+        doc = Nokogiri::HTML(html)
+        li = doc.at_css('ol li')
+        reference = li.children.to_html if li.present? # extract just the reference from <li>
+        accumulator << reference.to_s
+      end
+    end
+
+    # Druids are kept as tags (keywords) in the BibTeX::Entry
+    def extract_bibtex_related_document_ids
+      lambda do |record, accumulator, _context|
+        record.keywords.to_s.split(',').map(&:strip).each do |druid|
+          accumulator << druid if druid =~ Exhibits::Application.config.druid_regex
+        end
+      end
+    end
+
     ##
     # Traject macro for converting BibTeX fields into Solr fields
     #
     # @param [Symbol] `key` the BibTeX field name
-    def from_bibtex(key) # rubocop: disable Metrics/AbcSize, Metrics/CyclomaticComplexity
+    def extract_bibtex_field(key) # rubocop: disable Metrics/AbcSize, Metrics/CyclomaticComplexity
       # Note that `record` is a BibTeX::Entry object
       lambda do |record, accumulator, _context|
-        case key # BibTeX fields don't all follow the same rules, so handle them here
-        when :id # we need to transform the key for a good id
-          accumulator << record.key.to_s.gsub(%r{http:\/\/zotero.org\/groups\/\d*\/items\/}, '')
-        when :key # `key` is always available (yet `provides?(:key)` is false)
-          accumulator << record.key.to_s
+        case key.to_sym # BibTeX fields don't all follow the same rules, so handle them here
         when :type # conflates with BibTeX notion of type (`record.type`) rather than field "type"
           accumulator << record.fields[:type].to_s.presence if record.fields.include?(:type)
         when :ref_type # use record.type to get BibTeX type
           accumulator << BIBTEX_ZOTERO_MAPPING[record.type] if BIBTEX_ZOTERO_MAPPING.include?(record.type)
         when :year # built-in method in BibTeX::Entry
           accumulator << record.year
-        when :raw # serialization into BibTeX format
-          accumulator << record.to_s
         else # use the dynamically constructed BibTeX method to fetch the field rather than fetch/get
-          accumulator << record.send(key).to_s.presence if record.respond_to?(key)
+          if record.respond_to?(key.to_sym)
+            result = record.send(key.to_sym).to_s.presence
```

Comment:
below this line, add the following to enable extraction macros to run on the values array:

```
result = Macros::Extraction.apply_extraction_options(result, options)
```

This allows at least the cases that are _not_ handled in lines 71-77 to support the extraction additions. I think a general pattern to highlight with future traject use is to move these sorts of field specific handlers to separate methods to make it really clear they aren't available for similar macros usage.

Reply:
thank you!